raster-foundry/raster-foundry
# Enable RF/Planet data switch, planet data filter, and planet data browse

## Overview

This PR: 
- adds a Planet-RF dropdown to switch data repo
- refactors code in scene filter pane component to accommodate planet data filter
- refactors to one way data binding between scene filter pane component and scene browse
- enables adding planet token upon switching to browse Planet scenes if there is no such token on file
- makes sure filters correctly request RF scene lists and request empty planet scene list upon switching and applying filters (pulling planet scene lists using the already written planet service will be the next step)
~- lets the session store which data repo switch was on and the corresponding filters~

**Updates on 11/11/2017:**
- Hook up scene filters with planet service.
- Set map extent as bbox filter for planet - other filters are same as RF
- Reshape data returned from planet to approximate RF scene object data shape (further reshape may be necessary)
- List planet scenes on sidebar: 15 scenes at a time (which is the rate limit per sec to get thumbnails) in a batch of 250 scenes (max) per request result
- Enable requesting more planet scenes with the paginated link. Show number of scenes in current batches.

**Updates on 11/12/2017:**
- List planet thumbnails on sidebar.
- Add planet scene detail modal.
- Set planet scene tile and data footprint to multi-polygon to be compatible with image overlay service.
- Show planet scene filter fields and metadata on its detail modal.
- Show thumbnail preview on detail modal.
- Enable planet scene browse (hover and select).

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="1920" alt="planet scene detail modal" src="https://user-images.githubusercontent.com/16109558/32704229-c9f5ea98-c7cf-11e7-9c1a-9978765ff71d.png">

<img width="1920" alt="a bunch of planet scene thumbnails added to the map" src="https://user-images.githubusercontent.com/16109558/32704233-d530ac86-c7cf-11e7-9e48-9e5a2df38d1a.png">



### Notes

Will add a feature flag once the review of this PR is done. Will remove the feature flag after all related features are ready.


## Testing Instructions

 * Navigate to scene browse and open the filter pane.
 * Apply filters on RF data and see if it behaves as it used to.
 * Switch to planet data from the dropdown and apply filters. 

**Updates on 11/11/2017:**
 * Apply filters and check if scenes are being requested based on filter (250 scenes max per request) and if they are listed on the side bar (15 scenes are displayed on sidebar at a time). 
 * Try the pagination by clicking on the 'load more scenes button', see if more scenes are requested and the number of scenes on top is updated.
 ~* Please note that displaying their thumbnails and putting thumbnails on map (with other browsing related features) are in another issue, and will be implemented in the next commit in this PR.~

**Updates on 11/12/2017:**
 * Click on planet scene sidebar items and see if info and thumbnails are displayed correctly in the detail modal.
 * Hover on planet scene sidebar items and see if thumbnails are shown on the map.
 * Toggle the checkboxes of planet scene sidebar items and check if the thumbnails are correctly added/deleted from the map.
 * Please note that adding planet scenes to a project should not work at this point - this will be in another card.


Closes #2333

**Updates on 11/11/2017:**

Closes #2334

**Updates on 11/12/2017:**

Closes #2336
